From 6deab1ebda38bc870bcf9a12382ac74df0dd9c73 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 1 Feb 2023 17:34:10 +0100
Subject: [PATCH] debug: Fix build with gcc-13

* fixes:
../../git/src/base/debug/profiler.cc:91:1: error: 'ReturnAddressLocationResolver' does not name a type
../../git/src/base/debug/profiler.h:56:21: error: 'ReturnAddressLocationResolver' was not declared in this scope
../../git/src/base/debug/profiler.h:56:9: error: ISO C++ forbids declaration of 'uintptr_t' with no type [-fpermissive]
../../git/src/base/debug/profiler.h:56:9: error: typedef 'base::debug::uintptr_t' is initialized (use 'decltype' instead)
../../git/src/base/debug/profiler.h:69:13: error: 'ReturnAddressLocationResolver' does not name a type
../../git/src/base/debug/stack_trace.cc:390:62: error: 'size_t base::debug::TraceStackFramePointersFromBuffer(uintptr_t, uintptr_t, const void**, size_t, size_t, bool)' redeclared as different kind of entity
../../git/src/base/debug/stack_trace.h:185:5: error: 'uintptr_t' was not declared in this scope
../../git/src/base/debug/stack_trace.h:186:5: error: 'uintptr_t' was not declared in this scope
../../git/src/base/debug/stack_trace.h:187:5: error: expected primary-expression before 'const'
../../git/src/base/debug/stack_trace.h:188:12: error: expected primary-expression before 'max_depth'
../../git/src/base/debug/stack_trace.h:189:12: error: expected primary-expression before 'skip_initial'
../../git/src/base/debug/stack_trace.h:190:52: error: expression list treated as compound expression in initializer [-fpermissive]
../../git/src/base/debug/stack_trace.h:190:5: error: expected primary-expression before 'bool'
../../git/src/base/debug/stack_trace.h:59:13: error: 'uintptr_t' does not name a type

Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/chromium94/+/344617 debug: Fix build with gcc-13]

Change-Id: I6874468a606bc62c73fed66c5703cb8c02f3e405
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/base/debug/profiler.h    | 1 +
 src/base/debug/stack_trace.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/base/debug/profiler.h b/src/base/debug/profiler.h
index 1229e06234..5f15e2aa1f 100644
--- a/src/base/debug/profiler.h
+++ b/src/base/debug/profiler.h
@@ -7,6 +7,7 @@
 
 #include <stddef.h>
 
+#include <cstdint>
 #include <string>
 
 #include "base/base_export.h"
diff --git a/src/base/debug/stack_trace.h b/src/base/debug/stack_trace.h
index a3a713656c..2f77709d16 100644
--- a/src/base/debug/stack_trace.h
+++ b/src/base/debug/stack_trace.h
@@ -7,6 +7,7 @@
 
 #include <stddef.h>
 
+#include <cstdint>
 #include <iosfwd>
 #include <string>
 
